modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {Editor} from './Editor';
+
+import '../css/main.scss';
+
+const BalloonEditor = ({config = {}, contents, name, ...otherProps}) => {
+	const defaultExtraPlugins = 'balloontoolbar,floatingspace';
+
+	const getConfig = () => {
+		const extraPlugins = config.extraPlugins
+			? `${config.extraPlugins}`
+			: '';
+
+		return {
+			...config,
+			extraAllowedContent: '*',
+			extraPlugins: `${extraPlugins}${defaultExtraPlugins}`,
+		};
+	};
+
+	let initialCssClass;
+
+	return (
+		<Editor
+			config={getConfig()}
+			name={name}
+			onBeforeLoad={(CKEDITOR) => {
+				CKEDITOR.disableAutoInline = true;
+
+				initialCssClass = CKEDITOR.env.cssClass;
+
+				CKEDITOR.env.cssClass = 'lfr_balloon_editor';
```

Comment:
I'm wondering if this should be `CKEDITOR.env.cssClass += ' lfr_balloon_editor'` or `CKEDITOR.env.cssClass = `${CKEDITOR.env.cssClass} lfr_balloon_editor`.

Also I think our convention is to use dashes instead of underscores for our CSS classes. I'd consult an expert here, but we might want to make this `lfr-balloon-editor`. I know CKEditor's approach is different, but since this is our own code, I think it makes sense it follows our own rules.

Reply:
About the `CKEDITOR.env.cssClass`, you're right: it might be dangerous to replace it with our own class ( @ambrinchaudhary asked me the same thing via Slack) 

I'll got with dashes in the class name, and use our conventions as much as possible.